controller/controller_server/exception_handler.py

```diff
@@ -4,17 +4,29 @@
 
 import controller.array_action.errors as array_errors
 from controller.common.csi_logger import get_stdout_logger
-from controller.controller_server.errors import ValidationException
+from controller.controller_server.errors import ObjectIdError, ValidationException
 
 logger = get_stdout_logger()
 
 status_codes_by_exception = {
     ValidationException: grpc.StatusCode.INVALID_ARGUMENT,
     array_errors.IllegalObjectName: grpc.StatusCode.INVALID_ARGUMENT,
+    array_errors.IllegalObjectID: grpc.StatusCode.INVALID_ARGUMENT,
     array_errors.PoolParameterIsMissing: grpc.StatusCode.INVALID_ARGUMENT,
+    array_errors.UnsupportedConnectivityTypeError: grpc.StatusCode.INVALID_ARGUMENT,
+    array_errors.SnapshotSourcePoolMismatch: grpc.StatusCode.INVALID_ARGUMENT,
     array_errors.ObjectNotFoundError: grpc.StatusCode.NOT_FOUND,
+    array_errors.HostNotFoundError: grpc.StatusCode.NOT_FOUND,
+    array_errors.NoIscsiTargetsFoundError: grpc.StatusCode.NOT_FOUND,
+    array_errors.VolumeAlreadyUnmappedError: grpc.StatusCode.NOT_FOUND,
+    ObjectIdError: grpc.StatusCode.NOT_FOUND,
     array_errors.PermissionDeniedError: grpc.StatusCode.PERMISSION_DENIED,
-    array_errors.NotEnoughSpaceInPool: grpc.StatusCode.RESOURCE_EXHAUSTED
+    array_errors.ObjectIsStillInUseError: grpc.StatusCode.FAILED_PRECONDITION,
+    array_errors.VolumeMappedToMultipleHostsError: grpc.StatusCode.FAILED_PRECONDITION,
```

Comment:
this error seems to be relevant only to a specific method.
if so, then as I [wrote before](https://github.com/IBM/ibm-block-csi-driver/pull/305#discussion_r608638835), please ensure there are no "method-specific" errors here

Reply:
@ArbelNathan what do you think?

Reply:
well, what do you suggest?

Reply:
keep the try-except wherever there is uncommon exceptions to except, as `CreateVolume` is in current develop branch

Reply:
ok, sounds good

Reply:
@ArbelNathan, Ori's suggestion adds more lines and indentations to the already huge file.
This is the reason for removing it.
Do you still prefer his solution? If so, please explain.

Reply:
the `grpc StatusCode` we are returning are by spec, if a grpc is not supposed to return by spec, we should follow.

Reply:
After talking with Arbel, I decided to change the code. Currently, we have urgent things to do. Thus, I postpone this task